vowpalwabbit/parse_example_json.h

```diff
@@ -1726,11 +1732,28 @@ bool parse_line_json(vw* all, char* line, size_t num_chars, v_array<example*>& e
     VW::template read_line_decision_service_json<audit>(*all, examples, line, num_chars, false,
         reinterpret_cast<VW::example_factory_t>(&VW::get_unused_example), all, &interaction);
 
+    if (!interaction.eventId.empty())
+    {
+      if (all->example_parser->metrics->FirstEventId.empty())
```

Comment:
looks like metrics is initialized conditionally (based on boolean flag), but there is no similar checks here. Or am I missing something?

Reply:
yes, right now it is initialized in the same place where the bool of line 1726 of this file gets set. I was expecting a test or some binding to break this contract of not calling into set_json_reader( but doesn't seem like it. Anyways, do you think I should move the initialization over here or just check if its null or not?

Reply:
addressed this point, see latest diff